### PR TITLE
Autodoc feature and doc update

### PR DESCRIPTION
**Pull request by ezdac**
- This PR will make autodoc work and classes, methods and modules can now be referenced in the documentation as directives.
- Also, there is an automatic documentation of command-line options from `click`.
- Most of the README files were stripped down and the content transferred to the documentation.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -16,13 +16,13 @@
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
-#
 import os
 import sys
-sys.path.insert(0, os.path.abspath('../'))
+from recommonmark.parser import CommonMarkParser
```
Comment (loredanacirstea) [inline docs/conf.py]: Missing packages needed for docs building: `recommonmark`, `sphinx_autodoc_typehints`, `sphinx_rtd_theme`, `sphinx_click`
They should be added in `requirements-dev.txt`

File: docs/introduction.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+sends a single transaction with the last balance proof and his
+signature, to settle the debt. Also, gas cost is acceptable when the
+sender sends the balance proof along with the receiver's closing
+signature. Worst case scenario is the receiver closing the channel
+during the challenge period. Therefore, trustworthy sender-receiver
+relations are stimulated.
+
+Try out the µRaiden demo and build your own customized version,
+following our instructions below!
+target: https://travis-ci.org/raiden-network/microraiden
```
Comment (loredanacirstea) [inline docs/introduction.rst]: `target: https://travis-ci.org/raiden-network/microraiden` ?

File: docs/cmdline.rst [hunk 3]
```diff
@@ -0,0 +1,11 @@
+.. toctree::
+  :maxdepth: 2
+
+Command line options
+======================
+
+When invoking microraiden from the commandline, you have several options:
+
+.. click:: microraiden.click_helpers:main
```
Comment (loredanacirstea) [inline docs/cmdline.rst]: Docs show:
`microraiden [OPTIONS] COMMAND [ARGS]...` and 
`microraiden start [OPTIONS]`

Not valid commands at this point.

File: microraiden/microraiden/client/client.py [hunk 4]
```diff
@@ -29,6 +29,14 @@ def __init__(
             channel_manager_address: str = None,
             web3: Web3 = None
     ) -> None:
+        """
+
+        Args:
+            private_key:
```
Comment (loredanacirstea) [inline microraiden/microraiden/client/client.py]: `path to private key file of the proxy`

File: docs/api-reference.rst [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+directly.
+
+
+Client
+------------------
+
+.. autoclass:: microraiden.client.client.Client
+    :members:
+
+
```
Comment (loredanacirstea) [inline docs/api-reference.rst]: Any reason why only the `Client` api is here? I saw you created an entire API folder for autogenerated docs from docstrings.

**Comment by ezdac** [comment]
The changes requested here are still not yet implemented, but this will get handled by #437 once it's finished.

**Pull request closed without merging**